cachito/workers/pkg_managers/gomod.py

```diff
@@ -33,6 +35,39 @@
 MODULE_VERSION_RE = re.compile(r"/v\d+$")
 
 
+def run_download_cmd(cmd: Iterable[str], params: Dict[str, str]) -> str:
+    """Run gomod command that downloads dependencies.
+
+    Such commands may fail due to network errors (go is bad at retrying), so the entire operation
+    will be retried a configurable number of times.
+
+    The backoff is constant. The download commands are typically used for multiple dependencies
+    at the same time, and the failure can be caused by any of them. Exponential backoff would make
+    more sense if we were downloading dependencies individually.
+    """
+    n_tries = get_worker_config().cachito_gomod_download_max_tries
+
+    @backoff.on_exception(
+        backoff.constant,
+        CachitoRetcodeError,
+        jitter=None,  # use the constant 1s backoff rather than a random value between 0 and 1
+        max_tries=n_tries,
+        logger=log,
+    )
+    def run_go(_cmd, _params) -> str:
+        log.debug(f"Running {_cmd}")
+        return run_gomod_cmd(_cmd, _params)
+
+    try:
+        return run_go(cmd, params)
+    except CachitoRetcodeError:
+        err_msg = (
+            f"Processing gomod dependencies failed. Cachito tried the {' '.join(cmd)} command "
+            f"{n_tries} times. This may indicate a problem with your repository or Cachito itself."
```

Comment:
Or maybe `n_tries + 1` in this line?

Reply:
The `max_tries` argument is the total number of attempts, not the number of retries. If you use `max_tries=3`, then the function will be tried at most 3 times.